testing/src/test/java/io/stargate/it/cql/BoundStatementTest.java

```diff
@@ -0,0 +1,172 @@
+              .addPositionalValues(KEY, i)
+              .build());
+    }
+
+    // table with simple primary key, single cell.
+    session.execute("CREATE TABLE IF NOT EXISTS test2 (k text primary key, v0 int)");
+
+    // table with composite partition key
+    session.execute(
+        "CREATE TABLE IF NOT EXISTS test3 "
+            + "(pk1 int, pk2 int, v int, "
+            + "PRIMARY KEY ((pk1, pk2)))");
+  }
+
+  @Test
+  public void should_use_positional_values() {
+    PreparedStatement prepared = session.prepare("INSERT INTO test2 (k, v0) values (?, ?)");
+    session.execute(prepared.bind(KEY, VALUE));
+
+    Row row = session.execute("SELECT v0 FROM test2 WHERE k=?", KEY).one();
+    assertThat(row.getInt("v0")).isEqualTo(VALUE);
+  }
+
+  @Test
+  public void should_use_named_values() {
+    PreparedStatement prepared = session.prepare("INSERT INTO test2 (k, v0) values (:k, :v)");
+    session.execute(
+        prepared.boundStatementBuilder().setString("k", KEY).setInt("v", VALUE).build());
+
+    Row row = session.execute("SELECT v0 FROM test2 WHERE k=?", KEY).one();
+    assertThat(row.getInt("v0")).isEqualTo(VALUE);
+  }
+
+  @Test
+  public void should_allow_nulls_values() {
+    PreparedStatement prepared = session.prepare("INSERT INTO test2 (k, v0) values (?, ?)");
+    session.execute(prepared.bind(KEY, null));
+
+    Row row = session.execute("SELECT v0 FROM test2 WHERE k=?", KEY).one();
+    assertThat(row.isNull("v0")).isTrue();
+  }
+
+  @Test
+  @Disabled("Returns an empty result set instead of failing")
+  public void should_fail_if_missing_values() {
+    PreparedStatement prepared = session.prepare("SELECT v FROM test3 WHERE pk1=? and pk2=?");
+    assertThatThrownBy(() -> session.execute(prepared.bind(1)))
+        .isInstanceOf(InvalidQueryException.class);
+  }
+
+  @Test
+  @Disabled("Doesn't seem to work at the moment")
+  public void should_not_write_tombstone_if_value_is_implicitly_unset() {
+    PreparedStatement prepared = session.prepare("INSERT INTO test2 (k, v0) values (?, ?)");
+    session.execute(prepared.bind(KEY, VALUE));
+
+    BoundStatement boundStatement = prepared.bind(KEY);
+    assertThat(boundStatement.isSet("v0")).isFalse();
+    session.execute(boundStatement);
+
+    Row row = session.execute("SELECT v0 FROM test2 WHERE k=?", KEY).one();
+    assertThat(row.getInt("v0")).isEqualTo(VALUE);
+  }
+
+  @Test
+  @Disabled("Doesn't seem to work at the moment")
+  public void should_not_write_tombstone_if_value_is_explicitly_unset() {
+    PreparedStatement prepared = session.prepare("INSERT INTO test2 (k, v0) values (?, ?)");
+    BoundStatement boundStatement = prepared.bind(KEY, VALUE);
+    session.execute(boundStatement);
+
+    session.execute(boundStatement.unset("v0"));
+
+    Row row = session.execute("SELECT v0 FROM test2 WHERE k=?", KEY).one();
+    assertThat(row.getInt("v0")).isEqualTo(VALUE);
+  }
+
+  @Test
+  public void should_use_page_size_on_statement() {
```

Comment:
Codacy found an issue: [The JUnit 5 test method name 'should_use_page_size_on_statement' doesn't match '[a-z][a-zA-Z0-9]*'](https://app.codacy.com/gh/stargate/stargate/pullRequest?prid=6263918)